src/util/Xhr.js

```diff
@@ -133,9 +133,10 @@ class Xhr {
      *
      * @private
      * @param {string} [id] - Optional box item id
+     * @param {string} [fileId] - Optional box file id
      * @return {Promise} that resolves to a token
      */
-    getToken(id?: string): Promise<?string> {
+    getToken(id?: string, fileId?: string): Promise<?string> {
```

Comment:
There is already an id param, any need for another?

Reply:
for uploads, id is `folderId`. I need a way to pass `fileId`. It does look weird since `id` can be `fileId` in some other cases.

Reply:
ah maybe I could've just used the id prefix to tell if it's fileId or folderId, going to see if that works

Reply:
You could also create a new instance of XHR with the id being the file id

Reply:
I am wondering if one should change the signature of these methods here to take an item object. If its a file object, its parent can come from parent.id, hmm...